### PR TITLE
Add sensible heartbeat options when running in editor

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
When running in editor, prevent workers from disconnecting due to timeout. This allows a user to debug without effect.
This currently relies on `bridge_single_port_max_heartbeat_timeout_ms` which is only available in a custom runtime (single-port-timeout), and should only be merged when this has gone to faas.

#### Release note
Feature: Added `HeartbeatTimeoutWithEditorSeconds` and use it if WITH_EDITOR is defined to prevent workers disconnecting when debugging while running in editor.

#### Primary reviewers
@improbable-valentyn @aleximprobable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 1]
```diff
@@ -72,8 +79,10 @@ struct ConfigureConnection
 	FTCHARToUTF8 ProtocolLogPrefix;
 	Worker_ComponentVtable DefaultVtable{};
 	Worker_CompressionParameters EnableCompressionParams{};
-	Worker_KcpTransportParameters UpstreamParams{};
-	Worker_KcpTransportParameters DownstreamParams{};
+
+#if WITH_EDITOR
+	Worker_HeartbeatParameters HeartbeatParams{ WORKER_DEFAULTS_HEARTBEAT_INTERVAL_MILLIS, MAX_int64 };
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]: Looks like it's gonna take quite a while to timeout with `MAX_int64`? Is it safe to do so?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetConnection.cpp [hunk 2]
```diff
@@ -153,14 +153,19 @@ void USpatialNetConnection::InitHeartbeat(FTimerManager* InTimerManager, Worker_
 
 void USpatialNetConnection::SetHeartbeatTimeoutTimer()
 {
+	float Timeout = GetDefault<USpatialGDKSettings>()->HeartbeatTimeoutSeconds;
+#if WITH_EDITOR
+	Timeout = 10000.f;
```
Comment (aleximprobable) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetConnection.cpp]: Should the Editor timeout value be also stored in Settings?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetConnection.cpp [hunk 3]
```diff
@@ -153,14 +153,19 @@ void USpatialNetConnection::InitHeartbeat(FTimerManager* InTimerManager, Worker_
 
 void USpatialNetConnection::SetHeartbeatTimeoutTimer()
 {
+	float Timeout = GetDefault<USpatialGDKSettings>()->HeartbeatTimeoutSeconds;
+#if WITH_EDITOR
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetConnection.cpp]: This is going to happen even if you launch an editor build as a standalone, which might not always be desirable, e.g. when you want to test the gameplay logic surrounding players timing out.

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetConnection.cpp]: As per Alex's suggestion, I'll make this a configurable setting, so users can modify it if the circumstances require. Acceptable?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]: Yep, in this instance we never want to timeout.

**Comment by kevinimprobable** [comment]
Hey, is there anything holding this PR back from being merged?

**Pull request merged**